src/proc.h

```diff
@@ -540,6 +540,14 @@ class job_t : noncopyable_t {
 
     /// autocxx junk.
     RustFFIProcList ffi_processes() const;
+
+    /// autocxx junk.
+    const job_group_t &ffi_group() const;
+
+    /// autocxx junk.
+    /// The const is a lie and is only necessary since at the moment cxx's SharedPtr doesn't support
+    /// getting a mutable reference.
+    bool ffi_resume() const;
```

Comment:
This isn't really a limitation of cxx, it's a limitation (or rather, feature) of the rust language. A `shared_ptr<T>` is equivalent to an `Arc<T>` in rust, and you can't call an `&mut self` function with an `&T`. 

That means there's something potentially unsafe in the C++ code you call (the `resume()` function) that would need to be changed when that's ported to rust in order to use interior mutability so that it can be called with `&self` instead of `&mut self`. (I didn't look at `resume()` right now, it could be a perfect one-to-one translation of interior mutability just not marked `const`. I don't know.)

Reply:
Yeah, you are absolutely right. I didn't think about that, as the issue made it seem, that this should generally be possible.